### PR TITLE
[MoM] Psychic Animal Redux

### PR DESCRIPTION
**Pull request by Maleclypse**
#### Summary
Mods "Additional psychic fauna"

#### Purpose of change
Adds a couple of additional psychic fauna to MoM

#### Describe the solution
Birds man.  I feel possessed by the spirit of Alfred Hitchcock

#### Describe alternatives you've considered
Not psychic birds

### CODE REVIEW

**Comment by Maleclypse** [comment]
@Standing-Storm I'm wanting to give the pigeon passengers an area of effect aura that is provides a vitamin to any PC/NPCs inside the aura.  I want that effect to provide a vitamin to the characters slowly while they are in the aura and that vitamin as it increases increases strength while decreasing intelligence. I can make the vitamin effects work but I can't figure out how to make the aura/area of effect that impacts the pc whether they have tamed the pigeon passengers or not.  Any ideas?

File: data/mods/MindOverMatter/monsters/animal_psychics.json [hunk 1]
```diff
@@ -160,5 +209,83 @@
...
+        "monster_message": "%1$'s third eye opens and it coos!"
+      }
+    ],
+    "petfood": {
+      "food": [ "BIRDFOOD" ],
+      "feed": "The %s seems to like you!  It runs around your legs and seems friendly.",
+      "pet": "The %s runs around your leg."
+    },
+    "flags": [ "SEES", "HEARS", "SMELLS", "ANIMAL", "PATH_AVOID_DANGER_1", "WARM", "FLIES", "SWARMS", "CANPLAY", "KEEP_DISTANCE" ]
+  }
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/animal_psychics.json:257]:
```suggestion
  },
```

File: data/mods/MindOverMatter/monsters/monster_overrides.json [hunk 2]
```diff
@@ -107,6 +107,12 @@
     "type": "MONSTER",
     "regeneration_modifiers": [ [ "effect_vitakin_hurt", -1 ] ]
   },
+  {
+    "id": "mon_pigeon",
+    "copy-from": "mon_pigeon",
+    "type": "MONSTER",
+    "upgrades": { "half_life": 30, "into_group": "GROUP_PIGEON_UPGRADE" },
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/monster_overrides.json:114]:
```suggestion
    "upgrades": { "half_life": 30, "into_group": "GROUP_PIGEON_UPGRADE" }
```

File: data/mods/MindOverMatter/monsters/animal_psychics.json [hunk 3]
```diff
@@ -160,5 +209,63 @@
     },
     "extend": { "flags": [ "CANPLAY" ] },
     "upgrades": { "age_grow": 12, "into": "mon_cockatrice" }
+  },
+  {
+    "id": "mon_pigeon_passenger",
+    "type": "MONSTER",
+    "//": "Intentionally unable to be culled when friendly.",
+    "name": { "str": "pigeon passenger" },
+    "description": "A gentle, plump, small-billed bird with a third eye, between its bill and forehead.  On the ground, it struts about with a characteristic bobbing of the head.  It is a strong, swift flier thanks to its long wings and powerful flight muscles. Is it following you?",
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/animal_psychics.json:218]:
```suggestion
    "description": "A gentle, plump, small-billed bird with a third eye, between its bill and forehead.  On the ground, it struts about with a characteristic bobbing of the head.  It is a strong, swift flier thanks to its long wings and powerful flight muscles.  Is it following you?",
```

**Comment by Standing-Storm** [comment]
hmm...is that problem that the aura only applies when not tamed, only when tamed, or neither?

**Comment by Maleclypse** [comment]
> hmm...is that problem that the aura only applies when not tamed, only when tamed, or neither?

Haven't even started on the aura part not sure where to start with that lol.  I know what I want to do, I'm not sure if there is any way to do it beyond creating an invisible field they emit, but would that tank performance?

File: data/mods/MindOverMatter/monstergroups/monster_faction.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    "copy-from": "corvid",
+    "hates": "fear_hawk"
+  },
+  {
+    "type": "MONSTER_FACTION",
+    "name": "ratkin",
+    "copy-from": "ratkin",
+    "hates": "fear_hawk"
+  }
+]
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monstergroups/monster_faction.json:21]:
```suggestion
]

```

**Comment by Standing-Storm** [comment]
Another possibility would be the way I did dragon breath in #68035--give them a spell on a cooldown that fires and, it it hits a target, gives them an effect, then make casting the aura contingent on them having the effect. So if there's anyone nearby, they'll stack up the aura until the effect allowing them to cast it fades, but they won't be constantly trying to spam the aura in combat and it won't have the performance problems lots of fields do.

File: data/mods/MindOverMatter/monstergroups/monster_faction.json [hunk 5]
```diff
@@ -0,0 +1,21 @@
+[
+  {
+    "type": "MONSTER_FACTION",
+    "//": "decided to give them some personality",
+    "name": "fear_hawk",
+    "base_faction": "mutant",
+    "hates": [ "corvid", "ratkin" ]
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monstergroups/monster_faction.json:7]:
```suggestion
    "hate": [ "corvid", "ratkin" ]
```

File: data/mods/MindOverMatter/monstergroups/monster_faction.json [hunk 6]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  {
+    "type": "MONSTER_FACTION",
+    "name": "corvid",
+    "copy-from": "corvid",
+    "hates": "fear_hawk"
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monstergroups/monster_faction.json:13]:
```suggestion
    "hate": "fear_hawk"
```

File: data/mods/MindOverMatter/monstergroups/monster_faction.json [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  {
+    "type": "MONSTER_FACTION",
+    "name": "ratkin",
+    "copy-from": "ratkin",
+    "hates": "fear_hawk"
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monstergroups/monster_faction.json:19]:
```suggestion
    "hate": "fear_hawk"
```

File: data/mods/MindOverMatter/monsters/monster_special_attacks.json [hunk 8]
```diff
@@ -13,6 +13,20 @@
...
+    "id": "psychic_shriek",
+    "cooldown": 5,
+    "move_cost": 100,
+    "range": 3,
+    "damage_max_instance": [ { "damage_type": "psi_telepathic_damage", "amount": 10 } ],
+    "hit_dmg_u": "%1$s shrieks at your %2$s!",
+    "hit_dmg_npc": "%1$s shrieks at <npcname>!",
+    "no_dmg_msg_u": "%1$s's shriek radiates out at you without doing any damage.",
+    "no_dmg_msg_npc": "%1$s's shriek radiates out at <npcname> without doing any damage.",
+    "miss_msg_u": "%s's shriek radiates at you, but you dodge!",
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/monster_special_attacks.json:28]:
```suggestion
    "miss_msg_u": "%s's shriek radiates at you, but you dodge!"
```

File: data/mods/MindOverMatter/effects/effects_monster.json [hunk 9]
```diff
@@ -190,6 +190,21 @@
   },
   {
     "type": "effect_type",
+    "id": "pigeon_aura",
+    "name": [ "" ],
+    "apply_message": [ [ "" ] ],
+    "decay_messages": [ [ "" ] ],
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/effects/effects_monster.json:196]:
```suggestion
    "decay_messages": [ [ "" ] ],
```

File: data/mods/MindOverMatter/effects/effects_monster.json [hunk 10]
```diff
@@ -190,6 +190,20 @@
   },
   {
     "type": "effect_type",
+    "id": "pigeon_aura",
+    "name": [ "" ],
+    "apply_message": [ [ "" ] ],
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/effects/effects_monster.json:195]:
```suggestion
```

File: data/mods/MindOverMatter/effects/effects_monster.json [hunk 11]
```diff
@@ -190,6 +190,19 @@
...
+  },
+  {
+    "type": "effect_type",
+    "id": "pigeon_aura_prep",
+    "rating": "neutral",
+    "blood_analysis_description": "Patient has allergens in blood stream.",
+    "base_mods": { "cough_chance": [ 2 ], "perspiration_chance": [ 1 ] }
+  },
+  {
     "id": "effect_feral_regeneration",
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/effects/effects_monster.json:206]:
```suggestion
    "type": "effect_type",
    "id": "effect_feral_regeneration",
```

File: data/mods/MindOverMatter/monsters/monster_eoc_spells.json [hunk 12]
```diff
@@ -93,5 +93,19 @@
     "min_damage": 1,
     "max_damage": 2,
     "min_range": 30
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "pigeon_aura",
+    "condition": { "u_is_character": true },
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/monster_eoc_spells.json:100]:
```suggestion
    "condition": { "u_is_character" },
```

File: data/mods/MindOverMatter/monsters/monster_eoc_spells.json [hunk 13]
```diff
@@ -93,5 +93,19 @@
     "min_damage": 1,
     "max_damage": 2,
     "min_range": 30
+  },
+  {
+    "type": "effect_on_condition",
+    "id": "pigeon_aura",
+    "condition": { "u_is_character" },
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/monster_eoc_spells.json:100]:
```suggestion
    "condition": "u_is_character",
```

**Comment by Maleclypse** [comment]
@Standing-Storm did you move this ` delete_nullifying_fields_monster (spell_type) has two definitions from the same source (mindovermatter)!` or have I duplicated it in conflict resolution?

**Comment by Standing-Storm** [comment]
> @Standing-Storm did you move this ` delete_nullifying_fields_monster (spell_type) has two definitions from the same source (mindovermatter)!` or have I duplicated it in conflict resolution?

I did--I moved it over to death_effects (since it's only used on zombie null death)

File: data/mods/MindOverMatter/monsters/monsters_spells.json [hunk 14]
```diff
@@ -31,6 +31,38 @@
...
+    "description": "This cleans up the nullifying fields after a zombie null dies.  It's a bug if you have it.",
+    "valid_targets": [ "ground" ],
+    "flags": [ "SILENT", "NO_EXPLOSION_SFX" ],
+    "effect": "ter_transform",
+    "effect_str": "ter_null_zombie_delete_fields",
+    "shape": "blast",
+    "max_level": 1,
+    "min_aoe": 12,
+    "max_aoe": 12
+  },
```
Comment (Maleclypse) [inline data/mods/MindOverMatter/monsters/monsters_spells.json:65]:
```suggestion
```

**Comment by Maleclypse** [comment]
@Standing-Storm please review.  Feel free to make any changes you want :)

File: data/mods/MindOverMatter/effects/effects_monster.json [hunk 15]
```diff
@@ -188,6 +188,20 @@
     },
     "vitamins": [ { "vitamin": "iron", "rate": [ [ -10, -10 ] ], "tick": [ "1 m" ] } ]
   },
+  {
+    "type": "effect_type",
+    "id": "pigeon_aura",
+    "name": [ "" ],
+    "blood_analysis_description": "Blood sample contains psionic steroid particles and pigeon dander.",
```
Comment (Standing-Storm) [inline data/mods/MindOverMatter/effects/effects_monster.json:195]:
```suggestion
    "blood_analysis_description": "Blood sample contains noetic steroid particles and pigeon dander.",
```

File: data/mods/MindOverMatter/monsters/animal_psychics.json [hunk 16]
```diff
@@ -42,6 +42,55 @@
...
+    "dissect": "dissect_bird_sample_single",
+    "fear_triggers": [ "PLAYER_CLOSE" ],
+    "special_attacks": [
+      {
+        "type": "monster_attack",
+        "attack_type": "melee",
+        "id": "psychic_shriek",
+        "cooldown": 5,
+        "move_cost": 100,
+        "range": 3,
```
Comment (Standing-Storm) [inline data/mods/MindOverMatter/monsters/animal_psychics.json:81]:
```suggestion
        "range": 3,
        "condition": { "not": { "u_has_effect": "effect_psi_null" } },
```

File: data/mods/MindOverMatter/monsters/monsters_spells.json [hunk 17]
```diff
@@ -31,6 +31,24 @@
...
+    "id": "pigeon_aura",
+    "type": "SPELL",
+    "name": "Pigeon aura",
+    "description": "You consume human blood to keep the thirst at bay.",
+    "valid_targets": [ "self" ],
+    "effect": "effect_on_condition",
+    "effect_str": "pigeon_aura",
+    "shape": "blast",
+    "energy_source": "NONE",
+    "skill": "deduction",
```
Comment (Standing-Storm) [inline data/mods/MindOverMatter/monsters/monsters_spells.json:44]:
```suggestion

```

File: data/mods/MindOverMatter/monsters/monsters_spells.json [hunk 18]
```diff
@@ -31,6 +31,24 @@
     "min_range": 10,
     "max_range": 10
   },
+  {
+    "id": "pigeon_aura",
+    "type": "SPELL",
+    "name": "Pigeon aura",
+    "description": "You consume human blood to keep the thirst at bay.",
```
Comment (Standing-Storm) [inline data/mods/MindOverMatter/monsters/monsters_spells.json:38]:
```suggestion
    "description": "Anyone nearby will get stronger.",
```
Those are some hardcore pigeons

**Review by Standing-Storm — CHANGES_REQUESTED** [review]
Just a couple changes but looks great! I especially love the pigeons--MoM needs more psionic animals that aren't predatory (either on humans or on other animals).

File: data/mods/MindOverMatter/monsters/monsters_spells.json (hunk 18 above)
Comment (Maleclypse) [reply data/mods/MindOverMatter/monsters/monsters_spells.json:38]: Oops thanks for catching that!

**Comment by prharvey** [comment]
This is breaking the build: `mon_pigeon` is duplicated.

**Comment by BrettDong** [comment]
I'm puzzled how basic build test passed here while `mon_pigeon` is duplicated in `data/mods/MindOverMatter/monsters/monster_overrides.json` and `data/mods/MindOverMatter/monsters/monster_overrides_tameable.json`.

**Comment by Standing-Storm** [comment]
Fix up: #70487

**Comment by Maleclypse** [comment]
> I'm puzzled how basic build test passed here while `mon_pigeon` is duplicated in `data/mods/MindOverMatter/monsters/monster_overrides.json` and `data/mods/MindOverMatter/monsters/monster_overrides_tameable.json`.

Basic build test passed at least a week ago before the monster_overrides_tameable.json even existed.  So the Tameable monsters PR was merged this PR suffered no-conflicts and tests weren't rerun. Simple timing.

**Pull request merged**